Examples/Io/Root/include/ActsExamples/Io/Root/RootNTupleReader.hpp

```diff
@@ -0,0 +1,142 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Propagator/MaterialInteractor.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/Framework/IReader.hpp"
+#include "ActsExamples/Framework/IService.hpp"
+#include "ActsExamples/Framework/ProcessCode.hpp"
+
+#include <mutex>
+#include <vector>
+
+class TChain;
+
+namespace ActsExamples {
+
+class RootNTupleReader : public ActsExamples::IReader {
```

Comment:
Actually now that I look at it I'm a bit unhappy with the name, as it does not express the content of the N tuple... Or is in our context RootNTuple a common term that differs from the [ROOT type](https://root.cern.ch/doc/master/classTNtuple.html)?

Reply:
I guess it should be a track param reader so maybe something like `RootTrackParameterReader` ?